minet/youtube/client.py

```diff
@@ -132,54 +125,83 @@ def forge_replies_url(key, comment_id, token=None):
 
 class YouTubeAPIClient(object):
     def __init__(self, key, before_sleep_until_midnight=None):
-        self.key = key
+        if not isinstance(key, list):
+            key = [key]
+        self.keys = {k: True for k in key}
+        self.current_key = key[0]
         self.pool = create_pool()
         self.before_sleep = before_sleep_until_midnight
         self.retryer = create_request_retryer()
 
     @retrying_method()
     def request_json(self, url):
-        err, response, data = request_json(url, pool=self.pool)
 
-        if err:
-            raise err
+        while True:
+
+            url = url.replace("|", "&key=%s" % self.current_key)
```

Comment:
Okay I see now. Just append using `+` or `+=` no need for replace. What's more it could have unforseen consequences to replace pipe chars.